src/picasso-definition/components/heat-map/index.js

```diff
@@ -1,18 +1,13 @@
 import KEYS from '../../../constants/keys';
 import createBrush from '../../brush';
 
-export default function createHeatMap({ models, flags }) {
-  const { layoutService, chartModel } = models;
+export default function createHeatMap(chartModel) {
   const viewHandler = chartModel.query.getViewHandler();
   const { transform } = viewHandler;
   const dataHandler = chartModel.query.getDataHandler();
   let binWidthPx;
   let binHeightPx;
 
-  if (!(layoutService.meta.isBigData && flags.isEnabled('DATA_BINNING'))) {
```

Comment:
But isn't it safer / more efficient to just not create the component if the flag is not even on? Just an unnecessary component running its show function otherwise.

Reply:
For some reason, with this `if (!(layoutService.meta.isBigData && flags.isEnabled('DATA_BINNING')))` check, labels are misplaced(one step behind bins position) after doing selection to point and clear selection.

The component seems only create once. Not sure the exact reason why return false will causing this issue. Do you have any ideas what might causing the issue?

isBinnedData check already counts flag and isBigData. I think show function only runs when value changes or I am wrong here?

Reply:
I guess that depends on how `layoutService.meta.isBigData` is set? Sounds more like something is wrong in the flow, and should probably be fixed.

But I cannot see how:

```js
if (!flags.isEnabled('DATA_BINNING')) {
  return false;
}
```
Could possibly break something with bin labels? Because if the flag is not on, there should be no bins? Unless I'm misunderstanding.

Reply:
Talked with Niclas, he will help to see if can find the root cause. I will merge the PR first to solve label issue.